src/hotspot/share/runtime/arguments.cpp

```diff
@@ -862,13 +862,31 @@ static bool set_numeric_flag(JVMFlag* flag, char* value, JVMFlagOrigin origin) {
     if (is_neg) {
       int_v = -int_v;
     }
+    if ((!is_neg && v > max_jint) || (is_neg && -(intx)v < min_jint)) {
+      return false;
+    }
     return JVMFlagAccess::set_int(flag, &int_v, origin) == JVMFlag::SUCCESS;
   } else if (flag->is_uint()) {
+    // -XX:ParallelGCThreads expects that a value of max_juint+1 (4294967296)
+    // will set ParallelGCThreads to 0.  So, allow values of max_juint+1.
+    if (v > (julong)max_juint + 1) {
+      return false;
+    }
```

Comment:
This seems very suspicious. Where is the code that depends on this?

Reply:
Test test/hotspot/jtreg/gc/arguments/TestParallelGCThreads.java has the following code:

```
    // 4294967295 == (unsigned int) -1
    // So setting ParallelGCThreads=4294967295 should give back 4294967295
    // and setting ParallelGCThreads=4294967296 should give back 0. (SerialGC is ok with ParallelGCThreads=0)
    for (long i = 4294967295L; i <= 4294967296L; i++) {
      long count = getParallelGCThreadCount(
          "-XX:+UseSerialGC",
          "-XX:ParallelGCThreads=" + i,
          "-XX:+PrintFlagsFinal",
          "-version");
      Asserts.assertEQ(count, i % 4294967296L, "Specifying ParallelGCThreads=" + i + " does not set the thread count properly!");
    }
  }
```

Reply:
That test seems bizarre to me - perhaps someone from GC team can comment on why it expects to see what it sees. But I would not suggest we cripple the argument processing logic just because one test expects it to behave in a strange way.

Reply:
Setting ParallelGCThreads=4294967296 isn't a reasonable real-life use case. There's no need to check for the JVM's behavior under this situation (which now becomes illegal).

There's already a test for  "SerialGC is ok with ParallelGCThreads=0" on line 100 of this file. So the block quoted by Harold should be removed.